samples/Mvc.Client/Startup.cs

```diff
@@ -25,46 +29,51 @@ public class Startup {
         }
 
         public void ConfigureServices(IServiceCollection services) {
-            services.AddAuthentication();
-            services.AddMvc();
-
-            services.Configure<SharedAuthenticationOptions>(options => {
+            services.AddAuthentication(options => {
                 options.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
             });
+
+            services.AddMvc();
         }
 
         public void Configure(IApplicationBuilder app) {
             var factory = app.ApplicationServices.GetRequiredService<ILoggerFactory>();
             factory.AddConsole();
 
+            app.UseIISPlatformHandler();
+
+            app.UseForwardedHeaders(new ForwardedHeadersOptions {
+                ForwardedHeaders = ForwardedHeaders.All
+            });
+
             app.UseStaticFiles();
 
-            app.UseCookieAuthentication(options => {
-                options.AutomaticAuthenticate = true;
-                options.AutomaticChallenge = true;
-                options.AuthenticationScheme = CookieAuthenticationDefaults.AuthenticationScheme;
-                options.LoginPath = new PathString("/signin");
+            app.UseCookieAuthentication(new CookieAuthenticationOptions {
+                AutomaticAuthenticate = true,
+                AutomaticChallenge = true,
+                AuthenticationScheme = CookieAuthenticationDefaults.AuthenticationScheme,
+                LoginPath = new PathString("/signin")
```

Comment:
Yeah, but TBH, I'm not a huge fan of the implicit conversion because people are sometimes tempted to specify more than just a path (https://github.com/aspnet/Security/issues/334)